qiime2/metadata/tests/test_metadata_column.py

```diff
@@ -258,49 +258,45 @@ def test_case_insensitive_duplicate_ids(self):
 
 class TestSourceArtifacts(unittest.TestCase):
     def setUp(self):
+        self.md = Metadata(pd.DataFrame(
+            {'col': [1, 2, 3]}, index=pd.Index(['a', 'b', 'c'], name='id')))
         self.mdc = DummyMetadataColumn(pd.Series(
-            [1, 2, 3], name='col', index=pd.Index(['a', 'b', 'c'], name='id')))
+            [1, 2, 3], name='col', index=pd.Index(['a', 'b', 'c'], name='id')),
+            md=self.md)
 
-    def test_no_source_artifacts(self):
+    def test_no_source_artifact(self):
         self.assertEqual(self.mdc.artifacts, ())
 
-    def test_add_zero_artifacts(self):
-        self.mdc._add_artifacts([])
+    def test_source_artifact(self):
+        self.md._source_artifact = Artifact.import_data('Mapping', {'a': '1'})
+        self.assertEqual(self.md.artifacts, self.mdc.artifacts)
 
-        self.assertEqual(self.mdc.artifacts, ())
-
-    def test_add_artifacts(self):
-        # First two artifacts have the same data but different UUIDs.
-        artifact1 = Artifact.import_data('Mapping', {'a': '1', 'b': '3'})
-        self.mdc._add_artifacts([artifact1])
-
-        artifact2 = Artifact.import_data('Mapping', {'a': '1', 'b': '3'})
-        artifact3 = Artifact.import_data('IntSequence1', [1, 2, 3, 4])
-        self.mdc._add_artifacts([artifact2, artifact3])
-
-        self.assertEqual(self.mdc.artifacts, (artifact1, artifact2, artifact3))
-
-    def test_add_non_artifact(self):
-        artifact = Artifact.import_data('Mapping', {'a': '1', 'b': '3'})
-
-        with self.assertRaisesRegex(TypeError, "Artifact object.*42"):
-            self.mdc._add_artifacts([artifact, 42])
-
-        # Test that the object hasn't been mutated.
-        self.assertEqual(self.mdc.artifacts, ())
-
-    def test_add_duplicate_artifact(self):
-        artifact1 = Artifact.import_data('Mapping', {'a': '1', 'b': '3'})
-        artifact2 = Artifact.import_data('IntSequence1', [1, 2, 3, 4])
-        self.mdc._add_artifacts([artifact1, artifact2])
-
-        with self.assertRaisesRegex(
-                ValueError, "Duplicate source artifacts.*DummyMetadataColumn.*"
-                            "artifact: Mapping"):
-            self.mdc._add_artifacts([artifact1])
-
-        # Test that the object hasn't been mutated.
-        self.assertEqual(self.mdc.artifacts, (artifact1, artifact2))
+    def test_source_artifact_from_merge(self):
+        artifact1 = Artifact.import_data('Mapping', {'a': '1', 'b': '2'})
+        artifact2 = Artifact.import_data('Mapping', {'c': '3', 'd': '4'})
+        artifact3 = Artifact.import_data('Mapping', {'e': '5', 'f': '6'})
+
+        md1 = artifact1.view(Metadata)
+        md2 = artifact2.view(Metadata)
+        md3 = artifact3.view(Metadata)
+        merged_md = md1.merge(md2, md3)
+
+        mdc = merged_md.get_column('e')
+        self.assertEqual(mdc.artifacts, (artifact3,))
+
+#    Same as with test_metadata.py this isn't really a h=thing anymore
```

Comment:
As the comment says, same as the version of this test in `test_metadata.py`. Is this even relevant now?

Reply:
You could repurpose this test for the `merging` test suite --- worth testing that merging two artifact-based metadata explodes.